pkg/apis/resource/types.go

```diff
@@ -329,109 +841,64 @@ type PodSchedulingContextList struct {
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// ResourceClass is used by administrators to influence how resources
-// are allocated.
-//
-// This is an alpha type and requires enabling the DynamicResourceAllocation
-// feature gate.
-type ResourceClass struct {
+// DeviceClass is a vendor or admin-provided resource that contains
+// device configuration and selectors. It can be referenced in
+// the device requests of a claim to apply these presets.
+// Cluster scoped.
+type DeviceClass struct {
```

Comment:
I found one use for a `class.status`: we could add a standard `conditions` field there with one standardized condition "faulty" (name TBD) which gets set when one of the CEL expressions causes a runtime error.

We could do the same for `claim.status` and `claimtemplate.status`.

None of that has a chance of being in 1.31, but as we might add it later, perhaps it makes sense to already add the `class.spec` indirection now?

Reply:
I would *not* make the spec immutable. That breaks `kubectl apply -f driver-deployment.yaml` and any consumer of a class should be prepared for the class to get replaced *or* updated at any time.

Reply:
Agreed, let's add spec now. It's not immutable, but it *is* only taken into account at scheduling / allocation time. So, if it is changed, we're not going to reschedule anything according to the new criteria.

I would probably use `Ready` as the `ConditionType`. Or maybe `Valid`.